
**Task 8**

**Registration / referral ID - SIRSS2234**

**Full Name: Sumit Raosaheb Deshmukh** 


# **Importing required libraries**

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import matplotlib as mplr̥
import warnings
warnings.simplefilter("ignore")
%matplotlib inline

# **Data Description and Data Cleaning**

**Importing Data**

In [13]:
from google.colab import files
uploaded = files.upload()

Saving heart.csv to heart (1).csv


In [7]:
df = pd.read_csv('heart.csv') 

In [ ]:
df.head() #Displaying first five rows

In [ ]:
df.tail() #Displaying last five rows.

In [ ]:
print(f'In train data {df.shape[0]} Rows and {df.shape[1]} Columns') #Checking with Dimension of Data

In [ ]:
df.columns

In [ ]:
df.info() #Checking Data Types

**Checking For duplicate values**

In [ ]:
df.duplicated().sum()

**There is one duplicate value present**



In [15]:
df.drop_duplicates(inplace=True) #Droping duplicate value

In [ ]:
df.duplicated().sum()

**Checking for Missing Values**

In [ ]:
df.isnull().sum()

**The data has no missing values, so no further transformations are required.**

In [18]:
df.describe()  #Displaying Statastical Insights

age         sex          cp  ...         caa       thall      output
count  302.00000  302.000000  302.000000  ...  302.000000  302.000000  302.000000
mean    54.42053    0.682119    0.963576  ...    0.718543    2.314570    0.543046
std      9.04797    0.466426    1.032044  ...    1.006748    0.613026    0.498970
min     29.00000    0.000000    0.000000  ...    0.000000    0.000000    0.000000
25%     48.00000    0.000000    0.000000  ...    0.000000    2.000000    0.000000
50%     55.50000    1.000000    1.000000  ...    0.000000    2.000000    1.000000
75%     61.00000    1.000000    2.000000  ...    1.000000    3.000000    1.000000
max     77.00000    1.000000    3.000000  ...    4.000000    3.000000    1.000000

[8 rows x 14 columns]

**Now our Data is Clean We can do Further Analysis**

# **Analysis**

**Uni-variate Analysis**

In [19]:
plt.style.use('seaborn-darkgrid')

In [ ]:
plt.figure(figsize = (20,10))
sns.countplot(x = df['age'])
plt.title('Count of Patients Age', fontsize = 15)
plt.xlabel('Age', fontsize = 12)
plt.ylabel('Count', fontsize = 12)
plt.show()

In [ ]:
sns.displot(df['age'])
plt.title('Distribution of Age', fontsize = 15)
plt.xlabel('Age', fontsize = 12)
plt.ylabel('Count', fontsize = 12)
plt.show()

In [ ]:
sns.displot(df['trtbps'])
plt.title('Distribution of Blood Pressure around Patients', fontsize = 15)
plt.xlabel('Blood Pressure', fontsize = 12)
plt.ylabel('Count', fontsize = 12)
plt.show()

In [ ]:
sns.displot(df['chol'])
plt.title('Distribution of Cholestrol Level around Patients', fontsize = 15)
plt.xlabel('Cholestrol Level', fontsize = 12)
plt.ylabel('Count', fontsize = 12)
plt.show()

In [ ]:
sns.displot(x = 'thalachh' , data = df)
plt.title('Distribution of Heart Rate Around Patients' , fontsize= 15)
plt.xlabel('Heart Rate', fontsize = 12)
plt.ylabel('Count', fontsize = 12)

In [ ]:
import plotly.express as px
plt.figure(figsize = (10,10))
px.pie(df['sex'].value_counts().reset_index(), names = 'index', values = 'sex', title = 'Percentage of Male and Female Patients')


In [ ]:
plt.figure(figsize = (10,10))
sns.countplot(x = 'sex' , hue = 'output' ,  data = df)
plt.title('Distribution of Gender w.r.t. Heart Attack' , fontsize= 15)
plt.xlabel('Sex', fontsize = 12)
plt.ylabel('Count', fontsize = 12)

In [ ]:
plt.figure(figsize = (10,4))
sns.countplot(x = 'caa', hue = 'output', data = df)
plt.title('Distribution of no. of major blood vessels w.r.t. Heart Attack' , fontsize= 15)
plt.xlabel('Number of major vessels', fontsize = 12)
plt.ylabel('Count', fontsize = 12)

In [ ]:
plt.figure(figsize = (10,10))
sns.countplot(x = 'restecg', hue ='output', data = df)
plt.title('Distribution of no. of major blood vessels w.r.t. Heart Attack', fontsize= 15)
plt.xlabel('Resting Electrocardiographic Results', fontsize = 12)
plt.ylabel('Count', fontsize = 12)

In [ ]:
plt.figure(figsize = (10,10))
sns.barplot(x = 'cp', y = 'output', data= df);
plt.title('Distribution of Chest Pain w.r.t. Heart Attack' , fontsize= 15)
plt.xlabel('Chest Pain Type', fontsize = 9)
plt.ylabel('Output', fontsize = 9)

In [ ]:
plt.figure(figsize = (5,5))
sns.barplot(x = 'fbs' , y = 'output', data= df);
plt.title('Distribution of fasting Blood Sugar w.r.t. Heart Attack' , fontsize= 12)
plt.xlabel('Fasting Blood Sugar', fontsize = 9)
plt.ylabel('Output', fontsize = 9)

### **Conclusion of Uni-variate Analysis :**



***1.Most of the patients have age between 50 to 60, in which maximum number of patients have age 58.**

**2.Around 68.2% (207) are Male patients and 31.8% (96) are Female patients .**

**3.Most of the patients have 'Typical Angina' chest pain.**

**4.Most of the patients Blood Pressure lies between 120 to 140.**

**5.Most of the patients Cholestrol level lies between 200 to 260.**

**6.Most of the patients Heart Rate lies between 150 to 170.***




**Bi-Variate Analysis**

In [ ]:
plt.figure(figsize = (10,6))
sns.histplot(data = df, x = 'age', hue = 'output')
plt.title('Does age affect the heart rate ?' , fontsize = 15)
plt.xlabel('Age', fontsize = 12)
plt.ylabel('Count', fontsize = 12)
plt.show()

In [ ]:
plt.figure(figsize = (15,10))
sns.lineplot(x = 'age', y = 'trtbps', data = df)
plt.title('Blood Pressure with Age', fontsize = 15)
plt.xlabel('Age', fontsize = 12) 
plt.ylabel('Blood Pressure', fontsize = 12)
plt.show()

In [ ]:
plt.figure(figsize = (15,10))
sns.lineplot(x = 'age', y = 'chol', data = df)
plt.title('Cholestrol level with Age', fontsize = 15)
plt.xlabel('Age', fontsize = 12) 
plt.ylabel('Cholestrol Level', fontsize = 12)
plt.show()

In [ ]:
plt.figure(figsize = (15,10))
sns.lineplot(x = 'age', y = 'thalachh', data = df)
plt.title('Heart rate with Age', fontsize = 15)
plt.xlabel('Age', fontsize = 12) 
plt.ylabel('Heart Rate', fontsize = 12)
plt.show()


# **Conclusion of Bi-Variate Analysis :**

**1.There is not any impactabel relationship with age and heart attack.So we can't comment on either with Increasing the Age there is high Chance of Heart attack or Low Chance of Heart Attack.**

**2.There is high chance of Increase in Blood Pressure in the body With Increase in Age .**

**3.There is high chance of Increase in Cholestrol Level in the body with increase in Age.**

**4.There is high chance of Decrease in Heart Rate in the body with increase in Age.**


**Multi-Variate Analysis :**

In [ ]:
plt.figure(figsize = (15,10))
sns.lineplot(x = 'age', y = 'chol', hue = 'output', data = df)
plt.title('Chances of Heart attack with increase in Age and Cholestrol', fontsize = 15)
plt.xlabel('Age', fontsize = 12) 
plt.ylabel('Cholestrol Level', fontsize = 12)
plt.show()


In [ ]:
plt.figure(figsize = (20,10))
sns.lineplot(x = 'age', y = 'trtbps', hue = 'output', data = df)
plt.title('Chances of Heart attack with increase in Age and Blood Pressure', fontsize = 15)
plt.xlabel('Age', fontsize = 12) 
plt.ylabel('Blood Pressure', fontsize = 12)
plt.show()

In [ ]:

plt.figure(figsize = (20,10))
sns.lineplot(x = 'age', y = 'thalachh', hue = 'output', data = df)
plt.title('Chances of Heart attack with increase in Age and maximum heart rate', fontsize = 15)
plt.xlabel('Age', fontsize = 12) 
plt.ylabel('Heart Rate', fontsize = 12)
plt.show()

In [ ]:

df.hist(figsize = (20,10))
plt.show()

In [ ]:
df.corr() #Checking Co-Relation

In [ ]:
plt.figure(figsize = (20,9))
sns.heatmap(df.corr(), annot = True)


# **Conclusion of Multi-Variate Analysis**

**1.There is not any impactable relation between age and heart attack. So we can't say with Increasing the Age there is high Chance of Heart attack or Low Chance of Heart Attack.**

**2.There is high chance of Increase in Blood Pressure in the body With Increase in Age**.

**3.There is high chance of Increase in Cholestrol Level in the body with increase in Age.**

**4.There is high chance of Decrease in Heart Rate in the body with increase in Age**.

In [44]:
X_train = df.drop(['output'],axis =1).values
Y_train = df['output'].values

In [45]:

from sklearn.model_selection import train_test_split
x_train , x_test , y_train , y_test = train_test_split(X_train, Y_train, test_size = 0.20, random_state = 5)

In [ ]:
from sklearn.linear_model import LinearRegression
lin = LinearRegression()
lin.fit(x_train , y_train)

In [ ]:
lin.score(x_train , y_train)

In [ ]:
lin.score(x_test , y_test)

In [ ]:
lin.intercept_

In [ ]:

from sklearn.linear_model import LogisticRegression
lr = LogisticRegression()
lr.fit(x_train , y_train)

In [53]:
pred_train = lr.predict(x_train)
pred_test = lr.predict(x_test)

In [ ]:
from sklearn.metrics import accuracy_score
train_accuracy = accuracy_score(y_train, pred_train )
test_accuracy = accuracy_score(y_test, pred_test)
print("Training Accuracy: ", train_accuracy)
print("Testing Accuracy: ", test_accuracy)